lib/ansible/plugins/action/__init__.py

```diff
@@ -665,7 +665,9 @@ def _update_module_args(self, module_name, module_args, task_vars):
         module_args['_ansible_keep_remote_files'] = C.DEFAULT_KEEP_REMOTE_FILES
 
         # make sure all commands use the designated temporary directory if created
-        module_args['_ansible_tmpdir'] = self._connection._shell.tmpdir
+        # FIXME: deal with non root become, temporarily set to None and let remote_tmp take care of it
+        # module_args['_ansible_tmpdir'] = self._connection._shell.tmpdir
+        module_args['_ansible_tmpdir'] = None
```

Comment:
Heh, I just meant this code to show that setting _ansible_tmpdir = None was what we needed to do.  I think that it's a bit too big a change to set this to None unconditionally.  Probably better to revert the changes to how we call tempfile.* until we figure out how to implement this only for privilege descalation.

Reply:
just as easy to remove it, i left it like this to force fixing, which is what im looking into now, posted this JIC we want to go forward for RC, this is not final fix